### DISCUSSION TITLE
Is there a way to disable coil disk cache?

### DISCUSSION THREAD

**Discussion by halizton** [coil-kt/coil#1336] [Q&A] [answered]
In coil 2.0, if we do not specify a disk cache, then coil would use its own `RealDiskCache` even though we could set a disk cache into okhttp. What happens in the end is the coil internal `RealDiskCache` will come first over the okhttp's disk cache.

In my use case of migrating from Picasso to Coil, I noticed some image loading slowness in coil especially when the images are loaded from disk and had this discovery after diving into Coil source. In the end, I had to implement a dummy `coil.disk.DiskCache` and use it to replace the `RealDiskCache` in order to let coil use okhttp disk cache.

Put the performance between the okhttp disk cache and coil `RealDiskCache` apart, I think we need a way to disable the coil disk cache more explicitly in the public API in case people do want to use the okhttp disk cache for some reason. Or they just want to disable the disk cache at all for their use case or testing.

I am not sure if this design is intentional or it could improved.

**Comment by colinrtwhite** [comment, marked as answer]
You can set ImageLoader.Builder.diskCache(null) to disable the disk cache.

**Reply by halizton** [reply]
As mentioned earlier, if the `ImageLoader.Builder.diskCache` is null, when building the ImageLoader, the default `RealDiskCache` would be created and assigned to the ImageLoader. Please let me know if I missed anything.

https://github.com/coil-kt/coil/blob/481fa962f29c137eab07aca91944f783208acab7/coil-base/src/main/java/coil/ImageLoader.kt#L533

**Reply by colinrtwhite** [reply]
That’s not correct. Setting the disk cache to null creates a `lazyOf(null)` which is not null itself: https://github.com/coil-kt/coil/blob/481fa962f29c137eab07aca91944f783208acab7/coil-base/src/main/java/coil/ImageLoader.kt#L227

**Reply by halizton** [reply]
Oh, thanks, that's right. the diskCache is a `initializedLazyImpl` object after setting `ImageLoader.Builder.diskCache(null)`.
